### PR TITLE
Small fixes in averaging checkpoints

### PR DESCRIPTION
**Pull request by mravanelli**
In this pull request (PR), I've made some minor enhancements related to the management of checkpoint averages. Specifically:

1. I've introduced a unittest to validate its proper functioning.
2. I've modified the recipes in the train.py file that had the number of checkpoints to average hard-coded.

### CODE REVIEW

File: recipes/LibriSpeech/ASR/transducer/train.py [hunk 1]
```diff
@@ -274,11 +274,12 @@ def on_stage_end(self, stage, stage_loss, epoch):
                 train_stats=self.train_stats,
                 valid_stats=stage_stats,
             )
-            # We save multiple checkpoints as we will average them!
             self.checkpointer.save_and_keep_only(
                 meta={"WER": stage_stats["WER"], "epoch": epoch},
-                min_keys=["WER"],
+                min_keys=["ACC"],
+                num_to_keep=self.hparams.avg_checkpoints,
```
Comment (Adel-Moumen) [inline recipes/LibriSpeech/ASR/transducer/train.py:280]: we might need to retrain this model to better understand what it means in practice. Does it worsen the results or ?

**Comment by mravanelli** [comment]
Hi @Adel-Moumen,
I did the required fixes. As for the conformer transducer, I restored the number of averages to 1. I think, however, we can improve the performance with 5 or 10 averages just like other recipes. We can explore this in the future.

File: recipes/LibriSpeech/ASR/transducer/train.py [hunk 2]
```diff
@@ -274,11 +274,12 @@ def on_stage_end(self, stage, stage_loss, epoch):
                 train_stats=self.train_stats,
                 valid_stats=stage_stats,
             )
-            # We save multiple checkpoints as we will average them!
             self.checkpointer.save_and_keep_only(
                 meta={"WER": stage_stats["WER"], "epoch": epoch},
-                min_keys=["WER"],
+                min_keys=["ACC"],
```
Comment (Adel-Moumen) [inline recipes/LibriSpeech/ASR/transducer/train.py]: why did it changed from WER to ACC ?

**Review by Adel-Moumen — APPROVED** [review]
LGTM

**Pull request merged**